python/python/lance/vector.py

```diff
@@ -215,26 +217,37 @@ def compute_partitions(
             [pa.field("row_id", pa.uint64()), pa.field("partition", pa.uint32())]
         )
 
-        def _f() -> Iterable[pa.RecordBatch]:
+        def _partition_assignment() -> Iterable[pa.RecordBatch]:
             with torch.no_grad():
                 for batch in torch_ds:
                     batch: Dict[str, torch.Tensor] = batch
                     vecs = batch[column].reshape(-1, kmeans.centroids.shape[1])
-                    ids = batch["_rowid"].reshape(-1)
+                    ids = batch["_rowid"].reshape(-1).cpu().numpy()
```

Comment:
can we move this to after we call `kmeans.transform` to get some pipeing?

Reply:
Done